Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RazorTargetTests.cs

```diff
@@ -0,0 +1,160 @@
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Globalization;
+using System.IO;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarScanner.MSBuild.Common;
+using SonarScanner.MSBuild.Tasks;
+using SonarScanner.MSBuild.Tasks.IntegrationTests;
+using TestUtilities;
+
+namespace SonarScanner.Integration.Tasks.IntegrationTests.TargetsTests
+{
+    [TestClass]
+    public class RazorTargetTests
+    {
+        private const string ErrorLogFilePattern = "{0}.RoslynCA.json";
+
+        public TestContext TestContext { get; set; }
+
+        [TestMethod]
+        public void Razor_ForProductProject_CheckErrorLogProperties()
+        {
+            // Arrange
+            var rootInputFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "Inputs");
+
+            var projectSnippet = $@"
+<PropertyGroup>
+<ErrorLog>%%ErrorLogFile%%</ErrorLog>
+<SonarCompileErrorLog>%%ErrorLogFile%%</SonarCompileErrorLog>
+  <SonarQubeTempPath>{rootInputFolder}</SonarQubeTempPath>
+</PropertyGroup>
+";
+
+            var projectFilePath = CreateProjectFile(null, projectSnippet);
+
+            // Act
+            var result = BuildRunner.BuildTargets(TestContext,
+                projectFilePath,
+                TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
+
+            // Assert
+            result.AssertTargetExecuted(TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
+            AssertErrorLogIsSetBySonarQubeTargets(result);
+        }
+
+        private static void AssertErrorLogIsSetBySonarQubeTargets(BuildLog result)
+        {
+            var targetDir = result.GetCapturedPropertyValue(TargetProperties.TargetDir);
+            var targetFileName = result.GetCapturedPropertyValue(TargetProperties.TargetFileName);
+
+            var expectedErrorLog = Path.Combine(targetDir, string.Format(CultureInfo.InvariantCulture, ErrorLogFilePattern, targetFileName));
+
+            AssertExpectedErrorLog(result, expectedErrorLog);
+        }
+
+        private static void AssertExpectedErrorLog(BuildLog result, string expectedErrorLog)
+        {
+            result.AssertExpectedCapturedPropertyValue(TargetProperties.ErrorLog, expectedErrorLog);
+            result.AssertExpectedCapturedPropertyValue(TargetProperties.RazorSonarCompileErrorLog, expectedErrorLog);
+            result.AssertExpectedCapturedPropertyValue(TargetProperties.RazorCompilationErrorLog, expectedErrorLog);
+        }
+
+        /// <summary>
+        /// Creates a valid project with the necessary ruleset and assembly files on disc
+        /// to successfully run the "OverrideRoslynCodeAnalysisProperties" target
+        /// </summary>
+        private string CreateProjectFile(AnalysisConfig analysisConfig, string testSpecificProjectXml)
```

Comment:
It looks like that this piece of code is now 7 times in this project with some small variants. It might worth refactoring it out to a reusable method.

Reply:
Yeah that was in progress, i was just keen to have the feedback loop on this first iteration.